client-react/src/pages/app/deployment-center/container/DeploymentCenterContainerSource.tsx

```diff
@@ -42,8 +50,39 @@ const DeploymentCenterContainerSource: React.FC<DeploymentCenterFieldProps<Deplo
     },
   ];
 
+  const openConfigurationBlade = async () => {
+    const result = await portalContext.openBlade({
+      detailBlade: 'SiteConfigSettingsFrameBladeReact',
+      extension: 'WebsitesExtension',
+      detailBladeInputs: {
+        id: deploymentCenterContext.resourceId,
+      },
+    });
+
+    if (result) {
+      deploymentCenterContext.refresh();
+    }
+  };
+
+  useEffect(() => {
+    const isGitHubActionsOrContainerOnly = formProps.values.scmType === ScmType.GitHubAction || formProps.values.scmType === ScmType.None;
+    setShowBasicAuthError(
+      isGitHubActionsOrContainerOnly && !deploymentCenterPublishingContext.basicPublishingCredentialsPolicies?.scm.allow
+    );
+  }, [formProps.values.scmType, deploymentCenterPublishingContext.basicPublishingCredentialsPolicies?.scm.allow]);
```

Comment:
Use `useMemo` instead of `useEffect` and `useState`.

Add `useMemo` import in line 1's 'react' import.

```suggestion
  const showBasicAuthError = useMemo(() => {
    const isGitHubActionsOrContainerOnly = formProps.values.scmType === ScmType.GitHubAction || formProps.values.scmType === ScmType.None;
    return (
      isGitHubActionsOrContainerOnly && !deploymentCenterPublishingContext.basicPublishingCredentialsPolicies?.scm.allow
    );
  }, [formProps.values.scmType, deploymentCenterPublishingContext.basicPublishingCredentialsPolicies?.scm.allow]);
```